database/database_test.go

```diff
@@ -234,3 +236,423 @@ func TestStatesWithStatus(t *testing.T) {
+			{Name: "i.test", Policy: "bulk-1-year"},
+			{Name: "j.test", Policy: "bulk-18-week"},
+			{Name: "k.test", Policy: "bulk-1-year"},
+		},
+	},
+}
+
+// Test GetIneligibleDomainStates tests getting IneligibleDomainStates from the 
+// database
+func TestGetIneligibleDomainStates(t *testing.T) {
+
+	resetDB()
+
+	// domainStates should be empty as domains are not added to database
+	// test for entry that does not exist
+	domainStates, err := testDB.GetIneligibleDomainStates([]string{"a.test"})
+	if len(domainStates) != 0 {
+		t.Errorf("Empty database should contain no preloaded domains")
+	}
+
+	// add domains to the database
+	var statuses []string
+	statusReport := func(format string, args ...interface{}) {
+		formatted := fmt.Sprintf(format, args...)
+		statuses = append(statuses, formatted)
+	}
+	for _, tt := range getAndDeleteTests {
+
+		err := testDB.SetIneligibleDomainStates(
+			tt.wantStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+	}
+	// get domains from the database
+	for _, tr := range getAndDeleteTests {
+		domainStates, err = testDB.GetIneligibleDomainStates(tr.domainNames)
+		if err != nil {
+			t.Errorf("%s", err)
+		}
+
+		sort.Slice(domainStates, func(i, j int) bool { return domainStates[i].Name < domainStates[j].Name })
+		if len(domainStates) != len(tr.domainNames) {
+			t.Errorf("Incorrect count of states for test %s", tr.description)
+		}
+		for i, domainState := range domainStates {
+			if domainState.Name != (tr.domainNames[i]) {
+				t.Errorf("unexpected domain at position %d for test %s: %#v", i, tr.description, domainState)
+			}
+		}
+	}
+}
+
+// TestDeleteIneligibleDomainStates tests the DeleteIneligibleDomainStates function
+func TestDeleteIneligibleDomainStates(t *testing.T) {
+
+	resetDB()
+
+	// domainStates should be empty as domains are not added to database
+	domainStates, err := testDB.GetIneligibleDomainStates([]string{"a.test"})
+
+	if len(domainStates) != 0 {
+		t.Errorf("Empty database should contain no preloaded domains")
+	}
+
+	// add domains to the database
+	var statuses []string
+	statusReport := func(format string, args ...interface{}) {
+		formatted := fmt.Sprintf(format, args...)
+		statuses = append(statuses, formatted)
+	}
+	for _, tt := range getAndDeleteTests {
+		err := testDB.SetIneligibleDomainStates(
+			tt.wantStates,
+			statusReport,
+		)
+		if err != nil {
+			t.Errorf("[%s] cannot put states %s", tt.description, err)
+			return
+		}
+	}
+	// delete domains from the database
+	for _, tr := range getAndDeleteTests {
+		err = testDB.DeleteIneligibleDomainStates(tr.domainNames)
+		if err != nil {
+			t.Errorf("%s", err)
+		}
+	}
+
+	if len(domainStates) != 0 {
+		t.Errorf("Empty database should contain no preloaded domains")
+	}
+
+}
+
+// setIneligibleDomainTests is a struct that is used in testing SetIneligibleDomainStates
+var setGetDeleteDuplicateIneligibleDomainTests = []struct {
```

Comment:
I don't understand the difference between the two test table entries here. The test would be much easier to understand with only one (and thus no table).